gprofiler/profilers/dotnet.py

```diff
@@ -0,0 +1,127 @@
+
+
+@register_profiler(
+    "dotnet",
+    possible_modes=["dotnet-trace", "disabled"],
+    supported_archs=["x86_64"],
+    default_mode="dotnet-trace",
+)
+class DotnetProfiler(ProcessProfilerBase):
+
+    RESOURCE_PATH = "dotnet/tools/dotnet-trace"
+    MAX_FREQUENCY = 100
+    _EXTRA_TIMEOUT = 10
+
+    def __init__(
+        self,
+        frequency: int,
+        duration: int,
+        stop_event: Optional[Event],
+        storage_dir: str,
+        profile_spawned_processes: bool,
+        dotnet_mode: str,
+    ):
+        super().__init__(frequency, duration, stop_event, storage_dir)
+        assert (
+            dotnet_mode == "dotnet-trace"
+        ), "Dotnet profiler should not be initialized, wrong dotnet-trace value given"
+        self._metadata = DotnetMetadata(self._stop_event)
+
+    def _make_command(self, process: Process, duration: int, output_path: str) -> List[str]:
+        if duration > 3600 * 24:
+            raise ValueError("Duration exceeds one full day")
+        return [
+            resource_path(self.RESOURCE_PATH),
+            "collect",
+            "--format",
+            "speedscope",
+            "--process-id",
+            str(get_process_nspid(process.pid)),
+            "--profile",
+            "cpu-sampling",
+            "--duration",
+            str(datetime.timedelta(seconds=duration)),
+            "--output",
+            output_path,
+            #           str(self._frequency), - TODO: frequency handling to be determined
+        ]
+
+    def _profile_process(self, process: Process, duration: int, spawned: bool) -> ProfileData:
+        logger.info(
+            f"Profiling{' spawned' if spawned else ''} process {process.pid} with dotnet-trace",
+            cmdline=" ".join(process.cmdline()),
+            no_extra_to_server=True,
+        )
+        appid = None
+        app_metadata = self._metadata.get_metadata(process)
+        # had to change the dots for minuses because of dotnet-trace removing the last part in other case
+        local_output_path = os.path.join(self._storage_dir, f"dotnet-trace-{random_prefix()}-{process.pid}")
+        tempdir = f"/proc/{process.pid}/root/tmp"
+        with removed_path(local_output_path):
+            try:
+                run_process(
+                    self._make_command(process, duration, local_output_path),
+                    env={"TMPDIR": tempdir},
+                    stop_event=self._stop_event,
+                    timeout=self._duration + self._EXTRA_TIMEOUT,
+                    kill_signal=signal.SIGKILL,
+                )
+                local_output_path = local_output_path + ".speedscope.json"
+
+            except ProcessStoppedException:
+                raise StopEventSetException
+            logger.info(f"Finished profiling process {process.pid} with dotnet")
+            return ProfileData(
+                load_speedscope_as_collapsed(Path(local_output_path), self._frequency), appid, app_metadata
```

Comment:
Other profilers use `merge.parse_one_collapsed` which optionally adds the `comm` of the profiled process. I'd like the dotnet profiler to do so as well, otherwise as the first frame I get this
![image](https://user-images.githubusercontent.com/8831572/187089202-c13c4ca5-fd99-4218-85b1-aa48730c1c4a.png)

which isn't coherent with other top-level frames that are process `comm`s.

Let's add the same logic to `load_speedscope_as_collapsed` - pass `add_comm: str` which is prepended as first frame for each stack.